File: assign1/Ruby/src/perfect_number_imperative.rb
class PerfectNumberImperative

	def is_perfect(number)
		number > 1 && number * 2 == sum_of_factors(number)
	end
	
	def sum_of_factors(number)
		sum = 0
		for i in 1..number
			sum += i if number % i == 0
		end
		sum
	end
end

File: assign1/Java/src/perfectNumber/PerfectNumberImperative.java
package perfectNumber;

public class PerfectNumberImperative {
	
	public boolean isPerfect(int number) {
		return number > 1 && number * 2 == sumOfFactors(number); 
	}
	
	public int sumOfFactors(int number) {
		int sum = 0;
		
		for(int i = 1; i <= number; i++){
			if(number % i == 0) {
				sum += i;
			}
		}
		
		return sum;
	}
}
